deep-learning/_posts/2018-11-29-memoryGan.md

```diff
@@ -0,0 +1,132 @@
+optimization is used with 
+WGAN modifications. 
+
+![](/deep-learning/images/memorygan/eqn1.png)
+
+where $$L^G_{GAN}(\theta, s)$$ and $$\lambda_{CLS}L^G_{CLS}(\theta, s)$$ are GAN loss and cross-entropy 
+loss. $$y_c$$ is the one-hot encoding for the category c. 
+
+The category c is sampled uniformly and z is sampled from a Gaussian distribution. 
+
+The discriminator is trained according to the following equations:
+
+![](/deep-learning/images/memorygan/eqn2.png)
+
+__Sequential fine tuning__ 
+
+The authors define a sequence of T tasks, one for each category. They then train their GAN one task at a 
+time. Each GAN is initialized with the previous GAN's parameters.  
+
+![](/deep-learning/images/memorygan/eqn3.png)
+
+__Elastic Weight Consolidation__
+
+The authors present Elastic Weight Consolidation as a baseline for their new method to prevent forgetting. 
+Elastic Weight Consolidation involves adding regularization to prevent the GAN's parameters from changing 
+too much during each task. 
+
+![](/deep-learning/images/memorygan/eqn4.png)
+
+$$F_{t-1,i}$$ is the Fisher information matrix that indicates how sensitive the parameter $$\theta^G_{t,
+i}$$ is to forgeting.  
+
+
+## Method
+
+![](/deep-learning/images/memorygan/fig2.png)
+
+__Joint retraining with replayed samples__
+
+Their first method involves extending the current task's dataset with the memory from the previous task's 
+GAN. $$ S_t' = S_c \cup_{c\in{1,...,t-1}} \tilde{S_c}$$, where $$\tilde{S_c}$$ is the replay set generated by
+$$\tilde{x} = G_{\theta^G_{t-1}}(z, c)$$.
+
+![](/deep-learning/images/memorygan/eqn5.png)
+
+__Replay alignment__
+
+The second method the authors propose involves adding a loss term for the generator. This loss term is an 
+L2 loss between the current and previous GAN outputs for the same category c and latent vector z. 
+
+![](/deep-learning/images/memorygan/eqn6.png)
+
+
+## Results
+
+The authors tested their two MeRGANs on digits generation with Mnist and SVHN and scene generation with 4 
+classes from the LSUN dataset. They compared their results to joint training, sequential fine tuning, 
+Elastic Weight Consolidation and deep generative replay (Unconditional GAN with replay memory).
+
+# Mnist and SVHN
+![](/deep-learning/images/memorygan/fig3.png)
+
+![](/deep-learning/images/memorygan/fig4.png)
+
+The authors trained a classifier on real data and used the classification accuracy 
+on the generated data to evaluate forgetting.
+
+![](/deep-learning/images/memorygan/table1.png)
+
+# LSUN
+
+![](/deep-learning/images/memorygan/fig5.png)
+![](/deep-learning/images/memorygan/fig6.png)
+
+The authors used a classifier to evaluate forgetting. They tested a classifier trained on real data on generated data and a classifier trained on generated data on real data 
+(Rev acc.). Finally, they tested the Frechet Inception Distance (FID) on the generated data. 
```

Comment:
The sentence is cut into two halves by the image when rendering the file. Adding an additional LF after the period should fix it.